### ISSUE TITLE
Callback event on selection of Date in RangePicker

### ISSUE THREAD

**Issue by Gudivaka** [ant-design/ant-design#45611]
### What problem does this feature solve?

Let's say user wanted to make 00:00 selectable by default when they select Date. Currently there is no callback method which can identify user has selected the Date in Rangepicker.

When someone selecting the date of custom dates when we have presets in rangepicker, user wanted to get the startOftheday as selected when user selected the date. This help to set the time to startOftheday/endOftheDay of any time as per the user requirement whenever user selected the date.

Sample proposed Code added in below sandbox
https://codesandbox.io/s/range-picker-antd-5-10-3-forked-qmdsgc?file=/demo.tsx

### What does the proposed API look like?

onDateChange={(date, info) => {
        console.log(date, info);
        // date will have selected dates both start time and endtime
        // info will have user selected date of startTime or endTime
 }}

**Comment by zombieJ** [comment]
What's different with `onCalendarChange`?

**Comment by Gudivaka** [comment]
onCalendarChange callback is getting called only after user selecting the Ok button on range picker.. we wanted the callback assoonas user selects the date.. so that user selecting Oct 1st date.. we can set the time by default to startof the day

**Comment by Gudivaka** [comment]
Hope i was able explain the case.
Can we consider the request to work on the change to handle this case.?

**Comment by Gudivaka** [comment]
Hi @zombieJ .. can we consider this as feature request?

**Comment by zombieJ** [comment]
Yes. I'm working on this.

**Comment by zombieJ** [comment]
Picker component has been refactor in `5.14.0`, and this issue is discussed and handled with https://github.com/react-component/picker/pull/691 which will be marked as closed.

Please feel free to create new issue if you have any further question :)

Picker 组件于 `5.14.0` 版本进行了重构，此问题已被 https://github.com/react-component/picker/pull/691 中进行讨论与处理。

如果有任何问题，欢迎随时创建新的 issue :)
